sv.c

```diff
@@ -1417,12 +1414,15 @@ Perl_sv_upgrade(pTHX_ SV *const sv, svtype new_type)
         assert(new_type_details->body_size);
         /* We always allocated the full length item with PURIFY. To do this
            we fake things so that arena is false for all 16 types..  */
+#ifndef PURIFY
```

Comment:
It isn't clear to me from the commit message why this block and the one below newly become not-PURIFY only. Is that an additional bugfix, an additional optimization, or is it somehow newly required by the rest of this change?

Reply:
> It isn't clear to me from the commit message why this block and the one below newly become not-PURIFY only. Is that an additional bugfix, an additional optimization, or is it somehow newly required by the rest of this change?

It wasn't actually obvious to me either, coming back to this. (The branch is basically a group of cherry-picks of commits from early August). "newly required" is the answer, now documented in the commit message to 6d91a202e2c91c7b. As a side effect of you asking this, I spotted some more things I thought could be cleared up better, so they are also now in that commit.

Reply:
> "newly required" is the answer, now documented in the commit message to [6d91a20](https://github.com/Perl/perl5/commit/6d91a202e2c91c7bcafe475970b333db8863e7d7).

Cool, makes much more sense now thanks. In that commit message, though, `s/as/was/` in "the unreachable code as using".

Reply:
Well spotted. Fixed in the rebase that I just pushed